src/components/FieldSelector.tsx

```diff
@@ -14,9 +14,11 @@ export default class FieldSelector extends React.PureComponent<IProps, IState> {
     }
 
     render() {
-        const options = this.props.fields.map(
-            (field: data.IField, index: number) => <FieldSelectorOption field={field} key={index} />
-        );
+        let options = [];
+        for (let label in this.props.registry) {
```

Comment:
We should decouple the registry key from the UI label displayed in FieldSelectorOption